lightningd/peer_htlcs.c

```diff
@@ -1092,8 +1107,10 @@ void peer_got_commitsig(struct peer *peer, const u8 *msg)
 		  tal_count(failed), tal_count(changed));
 
 	/* New HTLCs */
-	for (i = 0; i < tal_count(added); i++)
-		added_their_htlc(peer, &added[i], &shared_secrets[i]);
+	for (i = 0; i < tal_count(added); i++) {
+		if (!peer_added_their_htlc(peer, &added[i], &shared_secrets[i]))
+			return;
```

Comment:
Not sure what happens in this case, is the message just left hanging and we don't proceed?

Reply:
Possibly.  The BOLT quoted in `peer_added_their_htlcs` says it SHOULD fail the channel if this fails, and it called `peer_internal_error` if it returned false at this point (which should reply with an `error` on all channels of that peer).  So we have replied: specifically, we have already replied with an error, and processing anything further is pointless since all channels with that peer will fail.

Reply:
Right, I forgot about the call to `peer_internal_error` above :+1: